Lib/posixpath.py

```diff
@@ -329,61 +308,55 @@ def expandvars(path):
             break
         i, j = m.span(0)
         name = m.group(1)
-        if name.startswith(start) and name.endswith(end):
+        if name[:1] == start and name[-1:] == end:
             name = name[1:-1]
         try:
             if environ is None:
                 value = os.fsencode(os.environ[os.fsdecode(name)])
             else:
                 value = environ[name]
-        except KeyError:
-            i = j
-        else:
             tail = path[j:]
             path = path[:i] + value
             i = len(path)
             path += tail
+        except KeyError:
+            i = j
     return path
 
 
 # Normalize a path, e.g. A//B, A/./B and A/foo/../B all become A/B.
 # It should be understood that this may change the meaning of the path
 # if it contains symbolic links!
 
+def _normpath_fallback(path):
```

Comment:
This is doing to lead to `os.path.normpath.__name__` being equal to `"_normpath_fallback"` in some situations, which seems like a regression. Does this change speed anything up?

Reply:
That's at the top of the pull request:

> Note: _normpath_fallback() has been exposed for benchmarking, that can be reverted when this is approved.

Reply:
Oh, just saw your note in the PR description about this. Sorry for not seeing that earlier :-) but you will need to remember to revert this change before the PR can be merged.

Reply:
Would you mind reverting this now, please? I'm struggling to follow what's changed otherwise. I trust your performance measurements are accurate :)

Reply:
That's fine (note that you can always hide whitespace differences).